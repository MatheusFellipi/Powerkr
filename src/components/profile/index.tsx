import { Avatar, Text, View } from "tamagui";

export default function Home() {
  return (
    <View>
      <Avatar
        circular
        size="$15"
      >
        <Avatar.Image
          accessibilityLabel="Cam"
          src="https://images.unsplash.com/photo-1548142813-c348350df52b?&w=150&h=150&dpr=2&q=80"
        />
        <Avatar.Fallback backgroundColor="$blue10" />
      </Avatar>
      <Text>Colaborador nome</Text>
      <Text>Departamento</Text>
    </View>
  );
}
