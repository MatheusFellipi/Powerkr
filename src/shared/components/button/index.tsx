import { Button, styled } from "tamagui";

export const BtnComponent = styled(Button, {
  fontSize: 14,
  borderRadius: 24,
  fontFamily: "InterBold",
  borderColor: "#4B5565",
  bg: "#202939",
  variants: {
    tabs: {
      activated: {},
      disabled: {
        borderColor: "$colorTransparent",
        bg: "$colorTransparent"
      }
    },
    styles: {
      login: {
        bg: "#F7B519",
        color: "#000"
      }
    }
  } as const
});
