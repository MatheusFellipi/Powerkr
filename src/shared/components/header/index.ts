import { Container } from "@shared/components/header/container";
import { Icon } from "@shared/components/header/icon";
import { TextComponent } from "@shared/components/textComponent";

export const HeaderComponent = {
  Text: TextComponent,
  Container: Container,
  Icon: Icon
};
