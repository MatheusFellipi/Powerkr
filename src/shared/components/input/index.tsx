import { useState } from "react";
import { Pressable, TextInputProps } from "react-native";
import { Icons } from "@assets/index";
import { Input, InputProps, Label, View } from "tamagui";

type Props = { label: string } & InputProps;

export const InputComponent = ({
  label,
  id,
  onChangeText,
  secureTextEntry,
  ...props
}: Props) => {
  const [showPassword, setShowPassword] = useState(secureTextEntry);
  return (
    <View>
      <Label
        color={"#CDD5DF"}
        htmlFor={id}
      >
        {label}
      </Label>
      <View
        fd="row"
        jc="space-between"
        ai="center"
        borderRadius={24}
        bg="#364152"
        borderWidth={1}
        borderColor="#4B5565"
      >
        <Input
          {...props}
          width={"90%"}
          borderRadius={24}
          secureTextEntry={showPassword}
          bg="$backgroundTransparent"
          borderColor="$backgroundTransparent"
        />
        {secureTextEntry && (
          <Pressable
            onPress={() => setShowPassword(!showPassword)}
            style={{
              width: "10%",
              paddingRight: 10
            }}
          >
            {showPassword ? (
              <Icons.EyeOn
                fill={"#A0AEC0"}
                height={25}
                width={25}
              />
            ) : (
              <Icons.EyeOff
                fill={"#A0AEC0"}
                height={25}
                width={25}
              />
            )}
          </Pressable>
        )}
      </View>
    </View>
  );
};
