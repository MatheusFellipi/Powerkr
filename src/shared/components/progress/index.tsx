import { CircularProgressComponent } from "@shared/components/progress/circular";

export const Progress = {
  Circular: CircularProgressComponent
};
