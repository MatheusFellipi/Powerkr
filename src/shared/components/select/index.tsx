import { useMemo, useState } from "react";
import { Icons } from "@assets/index";
import { Adapt, Select as S, Sheet } from "tamagui";

const items = [{ name: "Todos" }];

export function SelectComponent() {
  const [val, setVal] = useState("todos");
  return (
    <S
      id="food"
      value={val}
      onValueChange={setVal}
      disablePreventBodyScroll
    >
      <S.Trigger
        borderRadius={25}
        width={131}
        bg="$colorTransparent"
        borderColor="#4B5565"
        iconAfter={Icons.ArrowDown}
      >
        <S.Value placeholder="Something" />
      </S.Trigger>
      <Adapt
        when="sm"
        platform="touch"
      >
        <Sheet
          modal
          dismissOnSnapToBottom
          animationConfig={{
            type: "spring",
            damping: 20,
            mass: 1.2,
            stiffness: 250
          }}
        >
          <Sheet.Frame>
            <Sheet.ScrollView>
              <Adapt.Contents />
            </Sheet.ScrollView>
          </Sheet.Frame>
          <Sheet.Overlay
            animation="lazy"
            enterStyle={{ opacity: 0 }}
            exitStyle={{ opacity: 0 }}
          />
        </Sheet>
      </Adapt>
      <S.Content>
        <S.Viewport>
          <S.Group>
            {useMemo(
              () =>
                items.map((item, i) => {
                  return (
                    <S.Item
                      index={i}
                      key={item.name}
                      value={item.name.toLowerCase()}
                    >
                      <S.ItemText>{item.name}</S.ItemText>
                    </S.Item>
                  );
                }),

              [items]
            )}
          </S.Group>
        </S.Viewport>
      </S.Content>
    </S>
  );
}
