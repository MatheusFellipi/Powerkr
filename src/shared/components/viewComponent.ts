import { styled, View } from "tamagui";

export const ViewComponent = styled(View, {
  variants: {
    pin: {}
  } as const
});
