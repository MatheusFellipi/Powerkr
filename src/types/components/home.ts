type ApprovedProps = {
  id: number;
  title: string;
  date: Date;
  priority: string;
};

type MetricProps = {
  id: number;
  value: number;
  description: string;
};

type CardProfileProps = {
  name: string;
  company: string;
  urlImg: string;
};

type TabsComponentProps = {
  tab: string;
  handle: (value: string) => void;
};

type HomeComponentProps = {
  approved: ApprovedProps[];
  metric: MetricProps[];
  user: CardProfileProps;
};
