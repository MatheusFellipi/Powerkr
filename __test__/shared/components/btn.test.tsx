import { BtnComponent } from "@shared/components/button";
import { fireEvent, render, screen } from "@testing-library/react-native";
import { TamaguiProvider } from "tamagui";

import config from "../../../tamagui.config";

describe("<App />", () => {
  it("renders correctly", () => {
    const handleClick = jest.fn();
    const tree = render(
      <TamaguiProvider config={config}>
        <BtnComponent onPress={handleClick}>Test</BtnComponent>
      </TamaguiProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
    fireEvent.press(screen.getByText("Test"));
    expect(handleClick).toHaveBeenCalledTimes(1);
    expect(screen.getByText("Test")).toBeDefined();
  });
});
