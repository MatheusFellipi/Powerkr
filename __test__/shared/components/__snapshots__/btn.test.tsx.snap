// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<View
  accessibilityRole="button"
  cursor="pointer"
  focusable={true}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#202939",
      "borderBottomColor": "#4B5565",
      "borderBottomLeftRadius": 24,
      "borderBottomRightRadius": 24,
      "borderBottomWidth": 1,
      "borderLeftColor": "#4B5565",
      "borderLeftWidth": 1,
      "borderRightColor": "#4B5565",
      "borderRightWidth": 1,
      "borderStyle": "solid",
      "borderTopColor": "#4B5565",
      "borderTopLeftRadius": 24,
      "borderTopRightRadius": 24,
      "borderTopWidth": 1,
      "flexDirection": "row",
      "flexWrap": "nowrap",
      "height": 44,
      "justifyContent": "center",
      "paddingLeft": 18,
      "paddingRight": 18,
    }
  }
>
  <Text
    cursor="pointer"
    fontFamily="InterBold"
    lineBreakMode="clip"
    numberOfLines={1}
    style={
      {
        "color": "#fff",
        "flexGrow": 0,
        "flexShrink": 1,
        "fontFamily": "InterBold",
        "fontSize": 14,
        "fontWeight": "400",
        "letterSpacing": 0,
        "lineHeight": 23,
      }
    }
    suppressHighlighting={true}
    userSelect="none"
  >
    Test
  </Text>
</View>
`;
